fluentd-forward/teleport_client.go

```diff
@@ -0,0 +1,232 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"time"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+// TeleportSearchEventsClient is an interface for client.Client, required for testing
+type TeleportSearchEventsClient interface {
+	SearchEvents(ctx context.Context, fromUTC, toUTC time.Time, namespace string, eventTypes []string, limit int, startKey string) ([]events.AuditEvent, string, error)
+	Close() error
+}
+
+// TeleportClient represents wrapper around Teleport client to work with events
+type TeleportClient struct {
+	// client is an instance of GRPC Teleport client
+	client TeleportSearchEventsClient
+
+	// cursor current page cursor value
+	cursor string
+
+	// nextCursor next page cursor value
+	nextCursor string
+
+	// id latest event returned by Next()
+	id string
+
+	// pos current virtual cursor position within a batch
+	pos int
+
+	// batch current events batch
+	batch []events.AuditEvent
+
+	// batchSize is fetch batch size
+	batchSize int
+
+	// namespace is events namespace
+	namespace string
+
+	// types is events types list
+	types []string
+
+	// startTime is event time frame start
+	startTime time.Time
+}
+
+// NewTeleportClient builds Teleport client instance
+func NewTeleportClient(c *Config, cursor string, id string) (*TeleportClient, error) {
+	var cl *client.Client
+	var err error
+
+	if c.TeleportIdentityFile != "" {
+		cl, err = newUsingIdentityFile(c)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		cl, err = newUsingKeys(c)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	tc := TeleportClient{
+		client:    cl,
+		pos:       -1,
+		cursor:    cursor,
+		batchSize: c.BatchSize,
+		namespace: c.Namespace,
+		startTime: c.StartTime,
+	}
+
+	err = tc.fetch(id)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &tc, nil
+}
+
+// newUsingIdentityFile tries to build API client using identity file
+func newUsingIdentityFile(c *Config) (*client.Client, error) {
+	identity := client.LoadIdentityFile(c.TeleportIdentityFile)
+
+	config := client.Config{
+		Addrs:       []string{c.TeleportAddr},
+		Credentials: []client.Credentials{identity},
+	}
+
+	client, err := client.New(context.Background(), config)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return client, nil
+}
+
+// newUsingKeys tries to build API client using keys
+func newUsingKeys(c *Config) (*client.Client, error) {
+	config := client.Config{
+		Addrs: []string{c.TeleportAddr},
+		Credentials: []client.Credentials{
+			client.LoadKeyPair(c.TeleportCert, c.TeleportKey, c.TeleportCA),
+		},
+	}
+
+	client, err := client.New(context.Background(), config)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return client, nil
+}
```

Comment:
If you go with the above suggestion, you can just remove these helper functions.
```suggestion
```